ding/framework/middleware/pace_controller.py

```diff
@@ -0,0 +1,62 @@
+from typing import TYPE_CHECKING
+import time
+import math
+if TYPE_CHECKING:
+    from ding.framework import Task, Context
+
+
+def Pace_Controller(
+    task: "Task",
+    theme: str = "unknown",
+    active_state: bool = True,
+    filter_the_similar: bool = True,
+    start_up_delay: float = 0.5,
+    timeout: float = math.inf,
+    emit_option: bool = True
+):
+
+    if start_up_delay > 0:
+        time.sleep(start_up_delay)
+
+    random_event = "pace_control_{}".format(theme)
+    event_received = False
+    all_middleware_names = list(map(lambda fn: getattr(fn, "__name__", type(fn).__name__), task.middleware))
+
+    def _event_received(node_id, middleware_names):
+        nonlocal event_received
+        not_same = map(lambda name: name not in all_middleware_names, middleware_names)
+
+        if filter_the_similar:
+            not_same = any(not_same)
+        else:
+            not_same = True
+
+        if task.router.is_active:
+            if task.router.node_id != node_id and not_same:
+                event_received = True
+        else:
+            if not_same:
+                event_received = True
+
+    if active_state:
```

Comment:
Why we need this `active_state`?

Reply:
Yes, I guess you are right. For achieving a simple principle, we should not consider the behavior when pace controller is partly off.